lib/modules/ai_bot/commands/image_command.rb

```diff
@@ -53,55 +65,78 @@ def process(prompt:)
       engine = SiteSetting.ai_stability_engine
       api_url = SiteSetting.ai_stability_api_url
 
-      # API is flaky, so try a few times
-      3.times do
-        begin
-          thread =
-            Thread.new do
-              begin
-                results =
-                  DiscourseAi::Inference::StabilityGenerator.perform!(
-                    prompt,
-                    engine: engine,
-                    api_key: api_key,
-                    api_url: api_url,
-                  )
-              rescue => e
-                Rails.logger.warn("Failed to generate image for prompt #{prompt}: #{e}")
-              end
+      i = 0
+      threads =
+        prompts.map do |prompt|
+          seed = seeds ? seeds[i] : nil
+          i += 1
+          Thread.new(seed, prompt) do |inner_seed, inner_prompt|
+            attempts = 0
+            begin
+              DiscourseAi::Inference::StabilityGenerator.perform!(
+                inner_prompt,
+                engine: engine,
+                api_key: api_key,
+                api_url: api_url,
+                image_count: 1,
+                seed: inner_seed,
+              )
+            rescue => e
+              attempts += 1
+              retry if attempts < 3
+              Rails.logger.warn("Failed to generate image for prompt #{prompt}: #{e}")
+              nil
             end
-
-          show_progress(".", progress_caret: true) while !thread.join(2)
-
-          break if results
+          end
         end
+
+      while true
+        show_progress(".", progress_caret: true)
+        break if threads.all? { |t| t.join(2) }
       end
 
-      return { prompt: prompt, error: "Something went wrong, could not generate image" } if !results
+      results = threads.map(&:value).compact
+
+      if !results.present?
+        return { prompt: prompt, error: "Something went wrong, could not generate image" }
+      end
 
       uploads = []
 
-      results[:artifacts].each_with_index do |image, i|
-        f = Tempfile.new("v1_txt2img_#{i}.png")
-        f.binmode
-        f.write(Base64.decode64(image[:base64]))
-        f.rewind
-        uploads << UploadCreator.new(f, "image.png").create_for(bot_user.id)
-        f.unlink
+      i = 0
+      results.each do |result|
+        result[:artifacts].each do |image|
+          f = Tempfile.new("v1_txt2img_#{i}.png")
```

Comment:
You can do:

```ruby
Tempfile.create("v1_txt2img_#{i}.png") do |file|
  file.binmode
  file.write(Base64.decode64(image[:base64]))
  file.rewind
  // upload stuff
end
```

Then it is automatically closed and unlinked at the end of the block. Ruby docs recommend doing the `close` and `unlink` in an `ensure` block if not using `create` https://devdocs.io/ruby~3.2/tempfile#class-Tempfile-label-Good+practices

Reply:
will change!